reana_server/complexity.py

```diff
@@ -11,8 +11,37 @@
 from typing import Dict, List, Tuple
 
 from reana_commons.job_utils import kubernetes_memory_to_bytes
+from reana_commons.errors import REANAKubernetesMemoryLimitExceeded
 
-from reana_server.config import REANA_KUBERNETES_JOBS_MEMORY_LIMIT
+from reana_server.config import (
+    REANA_KUBERNETES_JOBS_MEMORY_LIMIT,
+    REANA_KUBERNETES_JOBS_MAX_USER_MEMORY_LIMIT,
+    REANA_KUBERNETES_JOBS_MAX_USER_MEMORY_LIMIT_IN_BYTES,
+)
+
+
+def validate_job_memory_limits(
+    workflow_name: str, complexity: List[Tuple[int, float]]
+) -> int:
+    """Validate that job memory limits does not exceed the maximum memory limit that users can assign to their job containers.
+
+    :param workflow_name: full workflow name including run number.
+    :param complexity: workflow complexity list which consists of number of initial jobs and the memory in bytes they require. (e.g. [(8, 1073741824), (5, 2147483648)])
+    :raises REANAKubernetesMemoryLimitExceeded: If workflow job memory limits exceed the maximum memory limit that users can assign to their job containers.
+    :return: maximal job memory (e.g. 1073741824)
+    """
+    if not complexity:
+        return 0
+    max_job_memory = max(complexity, key=lambda x: x[1])[1]
+
+    if (
+        REANA_KUBERNETES_JOBS_MAX_USER_MEMORY_LIMIT_IN_BYTES
+        and max_job_memory > REANA_KUBERNETES_JOBS_MAX_USER_MEMORY_LIMIT_IN_BYTES
+    ):
+        raise REANAKubernetesMemoryLimitExceeded(
+            f'The "kubernetes_memory_limit" provided in the workflow ({workflow_name}) exceeds the limit ({REANA_KUBERNETES_JOBS_MAX_USER_MEMORY_LIMIT}).'
+        )
+    return max_job_memory
```

Comment:
This function just validates and doesn't need to return any value. I think `return` can be removed.

Reply:
Function looks very weird without `return`, I've left `return None`

Reply:
*question(non-blocking):* What do you mean by "weird"?